### PR TITLE
Move the command line error to be printed LAST!

### PR DESCRIPTION
**Pull request by nh13**
@tfenne, this one's for you 🍺 

p0l0xvckvbluu

### CODE REVIEW

**Comment by yfarjoun** [comment]
Thanks! this annoys me SOO much!

File: src/main/java/picard/cmdline/CommandLineProgram.java [hunk 1]
```diff
@@ -262,10 +259,17 @@ protected boolean parseArgs(final String[] argv) {
...
         final String[] customErrorMessages = customCommandLineValidation();
         if (customErrorMessages != null) {
+            commandLineParser.usage(System.err, false);
+            if (customErrorMessages.length > 0) {
+                System.err.println(String.join("", Collections.nCopies(80, "#")));
+                System.err.println();
+            }
             for (final String msg : customErrorMessages) {
-                System.err.println(msg);
+                System.err.println("Error: " + msg + "\n");
```
Comment (tfenne) [inline src/main/java/picard/cmdline/CommandLineProgram.java]: I don't think you need the trailing `\n` since you're invoking `println`

**Comment by cmnbroad** [comment]
First of all - brilliant! Wish I'd thought of that.

But, I think @tfenne is right - to fully fix this we'd have to capture the PrintStream we give to the CLP onto a ByteArrayOutputStream and then write it to stderr after the CLP returns. The Barclay parser throws on errors, but I followed the same ordering in the catch clause in my PR. Anyway, if we don't do this here I'd be happy to fold it in as part of the Barclay PR, since it will save a lot of scrolling...

**Comment by nh13** [comment]
@cmnbroad if you can make it happen in Barclay, I'm all for closing this.

**Pull request closed without merging**